### DISCUSSION TITLE
Waybar broke after adding a wallpaper to a theme, reverting the file didn't fix it.

### DISCUSSION THREAD

**Discussion by vincentcomfy** [prasanthrangan/hyprdots#1876] [Q&A] [answered]
I know, I'm here a lot. I'm sorry.

Waybar started throwing errors after I followed the advice in [this thread](https://github.com/prasanthrangan/hyprdots/discussions/1783) where I tried to specify a directory for where themes could pull wallpapers from. Waybar immediately disappeared and after trying to start it manually in the terminal, I would get

```
❯ waybar
[2024-09-22 22:24:19.439] [info] Using configuration file /home/vincent/.config/waybar/config.jsonc
[2024-09-22 22:24:19.440] [info] Unable to receive desktop appearance: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop
[2024-09-22 22:24:19.440] [info] Using CSS file /home/vincent/.config/waybar/style.css
[2024-09-22 22:24:19.454] [warning] module backlight: Disabling module "backlight", No backlight found
[2024-09-22 22:24:19.456] [warning] No batteries.
[2024-09-22 22:24:19.492] [info] Bar configured (width: 3440, height: 29) for output: DP-2

** (waybar:6839): WARNING **: 22:24:19.493: Status Notifier Item with bus name ':1.33' and object path '/org/ayatana/NotificationItem/udiskie' is already registered
![240923_22h41m30s_screenshot](https://github.com/user-attachments/assets/f1f8485d-396a-4a82-ac45-574f46cd0954)

** (waybar:6839): WARNING **: 22:24:19.493: Status Notifier Item with bus name ':1.14' and object path '/org/ayatana/NotificationItem/nm_applet' is already registered
```

Waybar would still not show even despite running Hyde reload and rebooting my system.

I then somewhat fixed it by modifying the /home/vincent/.config/waybar/config.jsonc file (below) and removed an errant unclosed curly brace, but now my waybar doesn't have any of the customisation elements. It just shows my network information top right, clock in the middle and the open apps/windows in the top left. There are no theme/wallpaper buttons or the power button.

```
{ <-- I removed this.
    "clock": {
        "format": "{:%I:%M %p}",
        "rotate": "0",
        "format-alt": "{:%R 󰃭 %d·%m·%y}",
        "tooltip-format": "<span>{calendar}</span>",
        "calendar": {
            "mode": "month",
            "mode-mon-col": 3,
            "on-scroll": 1,
            "on-click-right": "mode",
            "format": {
                "months": "<span color='#ffead3'><b>{}</b></span>",
                "weekdays": "<span color='#ffcc66'><b>{}</b></span>",
                "today": "<span color='#ff6699'><b>{}</b></span>"
            }
        },
        "actions": {
            "on-click-right": "mode",
            "on-click-forward": "tz_up",
            "on-click-backward": "tz_down",
            "on-scroll-up": "shift_up",
            "on-scroll-down": "shift_down"
        }
    } 
}
```

How do I go about fixing this properly? Picture I think is attached for reference. The issue is that it _is_ technically working, but it doesn't look or behave like it used to and it bothers me. Shutting down or rebooting now requires me to use the terminal which is a little irritating.
![240923_22h41m30s_screenshot](https://github.com/user-attachments/assets/e1afc397-5286-456d-b3d9-cf070ada1d31)

**Comment by vincentcomfy** [comment]
So I looked at past threads, either I misunderstood the wallpaper thread or the way wallpapers are managed may have changed, but [according to this](https://github.com/prasanthrangan/hyprdots/discussions/913) it looks like we aren't supposed to modify the .jsonc file?

I tried following changing the waybar layout per [this thread](https://github.com/prasanthrangan/hyprdots/discussions/924), but it unfortunately didn't fix it (in fact, it made waybar disappear, again).

**Reply by kRHYME7** [reply]
Only change module inside ` ~/.config/waybar/modules/ `. Not e that do not follow linters on these files as it is not a complete jsonc format. 
Remember what module you change then check upstream what happen. 

> https://github.com/prasanthrangan/hyprdots/discussions/913 it looks like we aren't supposed to modify the .jsonc file?

To generate it again, run ` export reload_flag=1 && wbarconfgen `
also, please show the output for this

**Reply by vincentcomfy** [reply]
> Only change module inside `~/.config/waybar/modules/`. Not e that do not follow linters on these files as it is not a complete jsonc format. Remember what module you change then check upstream what happen.
> 
> > #913 it looks like we aren't supposed to modify the .jsonc file?
> 


Unfortunately I don't recall what was changed, however I can paste the output? https://pastebin.com/a1LyPxdb


> To generate it again, run `export reload_flag=1 && wbarconfgen` also, please show the output for this

`❯ export reload_flag=1 && wbarconfgen
zsh: command not found: wbarconfgen`

**Reply by vincentcomfy** [reply]
After looking at the .jsonc file again I'm not so sure I actually modified it now.

```
❯ cd ~/.config/hyde/themes
❯ ls
 'Catppuccin Latte'
 'Catppuccin Mocha'
 'Decay Green'
 'Edge Runner'
 'Frosted Glass'
 'Graphite Mono'
 'Gruvbox Retro'
 'Material Sakura'
 'Nordic Blue'
 'Rosé Pine'
 'Synth Wave'
 'Tokyo Night'
❯ cd Catppuccin\ Mocha
❯ ls
 hypr.theme
 kitty.theme
 kvantum
 rofi.theme
 wall.set -> '/home/vincent/.config/hyde/themes/Catppuccin Mocha/wallpapers'/20.12.09_a_hires.jpg
 wallpapers
 waybar.theme
```
```
❯ cd "Gruvbox Retro"
❯ ls
 hypr.theme
 kitty.theme
 kvantum
 rofi.theme
 wall.set -> '/home/vincent/.config/hyde/themes/Gruvbox Retro/wallpapers'/patterns.jpg
 wallpapers
 waybar.theme
```
Does anything look abnormal here? The wall.set is supposed to look like this? I am trying to figure out where I even modified the wallpaper and what changes I made. At this point I really need to start utilizing backups more. I even have a home server so there's no excuse really.

If you can help me figure this out, I'm going to start looking into backup solutions.

**Reply by kRHYME7** [reply]
> export reload_flag=1 && wbarconfgen zsh: command not found: wbarconfgen

my bad please add .sh. It should be wbarconfgen.sh


You can try to reboot. I reviewed the waybar output and it looks fine. But this is a common issue with waybar. There are times it's  so slow to load.

**Reply by vincentcomfy** [reply]
> > export reload_flag=1 && wbarconfgen zsh: command not found: wbarconfgen
> 
> my bad please add .sh. It should be wbarconfgen.sh

❯ export reload_flag=1 && wbarconfgen.sh
waybar: no process found

If I try and start waybar using "waybar" in the terminal:

```
❯ waybar
[2024-09-24 03:53:30.195] [info] Using configuration file /home/vincent/.config/waybar/config.jsonc
[2024-09-24 03:53:30.195] [error] Error parsing JSON: * Line 68, Column 1
  Missing '}' or object member name
```

Same issue as what I mentioned above where there's an unclosed squiggly brace.
Removing it and running waybar again:

```
❯ waybar
[2024-09-24 03:54:49.057] [info] Using configuration file /home/vincent/.config/waybar/config.jsonc
[2024-09-24 03:54:49.058] [info] Unable to receive desktop appearance: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.Settings” on object at path /org/freedesktop/portal/desktop
[2024-09-24 03:54:49.058] [info] Using CSS file /home/vincent/.config/waybar/style.css
[2024-09-24 03:54:49.072] [warning] module backlight: Disabling module "backlight", No backlight found
[2024-09-24 03:54:49.075] [warning] No batteries.

** (waybar:53117): WARNING **: 03:54:49.125: Status Notifier Item with bus name ':1.41' and object path '/org/ayatana/NotificationItem/udiskie' is already registered

** (waybar:53117): WARNING **: 03:54:49.125: Status Notifier Item with bus name ':1.15' and object path '/org/ayatana/NotificationItem/nm_applet' is already registered
[2024-09-24 03:54:49.139] [warning] Requested width: 29 is less than the minimum width: 56 required by the modules
[2024-09-24 03:54:49.139] [info] Bar configured (width: 56, height: 1440) for output: DP-2

** (waybar:53117): WARNING **: 03:54:49.161: Status Notifier Item with bus name ':1.41' and object path '/org/ayatana/NotificationItem/udiskie' is already registered

** (waybar:53117): WARNING **: 03:54:49.161: Status Notifier Item with bus name ':1.15' and object path '/org/ayatana/NotificationItem/nm_applet' is already registered
```

**Reply by kRHYME7** [reply]
Have you tried toggling the a different waybar layout. You can check the keybindings for that

**Reply by vincentcomfy** [reply]
> Have you tried toggling the a different waybar layout. You can check the keybindings for that

Yeah, unfortunately toggling the waybar layout is what made it go from incorrect to missing altogether. I mentioned it in a prior comment. Is there any way I can build from source or copy whatever the stock files are from github? It'd probably be a lot easier for me to go back to a known working configuration (make a backup!) and try the wallpaper again rather than try and undo whatever it is that I've broken.

**Reply by kRHYME7** [reply, marked as answer]
You can use our CLI here.

`Hyde check Config` You'll see the diff of the source and your local.

**Reply by vincentcomfy** [reply]
Thank you, so much. As always you managed to solve my issue.
![image](https://github.com/user-attachments/assets/ef5ab7bd-9747-4c2a-9c0d-8de6b813754f)

I promise I'll make backups before I start trying to break things again in the future.
